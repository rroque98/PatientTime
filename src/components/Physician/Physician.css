ul {
  padding: 0 0 0 2em;
}

.physician span {
  display: block;
  background-color: var(--notableBlue);
  color: var(--white);
}
