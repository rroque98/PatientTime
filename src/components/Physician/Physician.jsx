import React from "react";
import styled from "styled-components";

const AppointmentRow = styled.li`
  list-style-type: circle;
  line-height: 2em;
  color: ${props => props.theme.darkGrey};
  list-style-position: inside;
  text-indent: 1em;
  background-color: ${props => (props.select ? props.theme.notableBlue : null)};
`;

const ApptText = styled.span`
  color: ${props => (props.select ? props.theme.white : props.theme.black)};
`;

const Physician = ({ physician, updateApptsOnClick, currPhysId }) => {
  let classPhys = "";
  if (currPhysId === physician.id) {
    classPhys += "select";
  }

  return (
    <AppointmentRow
      select={classPhys}
      className={classPhys}
      onClick={e => updateApptsOnClick(e, physician.id)}
    >
      <ApptText select={classPhys}>
        {physician.lastName}, {physician.firstName}
      </ApptText>
    </AppointmentRow>
  );
};

export default Physician;
