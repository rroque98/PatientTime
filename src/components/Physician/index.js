export { default } from "./Physician.jsx";
