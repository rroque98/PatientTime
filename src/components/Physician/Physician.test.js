import React from "react";
import {
  render,
  fireEvent,
  cleanup,
  waitForElement
} from "react-testing-library";
import Physician from "./index.js";
import "jest-styled-components";

afterEach(cleanup);

const physician = {
  id: 1,
  lastName: "physLastName",
  firstName: "physFirstName"
};

// test("should apply physician class to li that is currently selected", () => {
//   const { getByText } = render(
//     <Physician physician={physician} currPhysId={1} />
//   );
//   const elem = getByText(`${physician.lastName}, ${physician.firstName}`);
//   expect(elem.parentNode.classList[0]).toBe("physician");
// });

// test("should not apply physician class to li that is not selected", () => {
//   const { getByText } = render(
//     <Physician physician={physician} currPhysId={2} />
//   );
//   const elem = getByText(`${physician.lastName}, ${physician.firstName}`);
//   expect(elem.parentNode.classList[0]).toBe(undefined);
// });

test("Physician snapshot", () => {
  const physComp = render(<Physician physician={{ id: 1, currPhysId: 1 }} />);
  expect(physComp).toMatchSnapshot();
  expect(physComp.firstChild).toMatchSnapshot();
});
