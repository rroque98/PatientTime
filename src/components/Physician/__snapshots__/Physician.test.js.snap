// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Physician snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  list-style-type: circle;
  line-height: 2em;
  list-style-position: inside;
  text-indent: 1em;
}

<body>
    <div>
      <li
        class="c0"
      >
        <span
          class=""
        >
          , 
        </span>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="sc-bdVaJa gVmRUz"
    >
      <span
        class="sc-bwzfXH jivAqU"
      >
        , 
      </span>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Physician snapshot 2`] = `undefined`;
